src/App.tsx

```diff
@@ -15,36 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_ALPHABETICALLY = 'Sort alphabetically',
+  SORT_BY_LENGTH = 'Sort by length',
+}
+
+interface SortedParams {
+  sortField: string;
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reversed }: SortedParams,
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SortType.SORT_ALPHABETICALLY) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField === SortType.SORT_BY_LENGTH) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
```

Comment:
```suggestion
    preparedGoods.reverse();
```
`reverse` mutates an array